### ISSUE TITLE
🚀 Feature Request: Provide a way to load variables/secrets from custom locations

### ISSUE THREAD

**Issue by alex35mil** [cloudflare/workers-sdk#4677]
### Describe the solution

Currently, there are two ways of defining env vars for workers in local env:
1. Hardcode them in `wrangler.toml`.
2. Define in the `.dev.vars` file.

Neither of those allows sharing configuration across services in a monorepo.

It would be great to have a flag that would allow loading env vars/secrets from custom env files:

```
wrangler dev --vars env/common.env --vars env/worker.env
```

**Comment by schardev** [comment]
+1. I have a monorepo setup (with `pnpm`) and I want to keep all my vars into a single `.env` file at the workspace root but loading that file inside the workers package is not possible currently. I have to keep a second `.env` file inside the workers package. It would be great if we can load vars from custom location.

**Comment by admah** [comment]
@alex35mil thanks for the feature request. 

In your example:

`wrangler dev --vars env/common.env --vars env/worker.env`

Are you thinking that the vars from `env/worker.env` would overwrite any duplicate keys found in `env/common.env`?

**Comment by schardev** [comment]
Yeah. I think it's very natural to assume that the vars from `env/worker.env` would overwrite the ones from `env/common.env`.

**Comment by alex35mil** [comment]
I don’t have overlapping values in such configs, but the described behavior makes sense.

**Comment by MrChrisRodriguez** [comment]
Has anyone found a workaround in the last 1.5y? Running into this issue currently and would rather avoid writing a custom script.

**Comment by penalosa** [comment]
This has now been implemented. You should be able to provide the `--env-file` argument to point to a custom env file.
